upgrade.go

```diff
@@ -74,13 +75,43 @@ func getVersionDiff(oldVersion, newversion string) (left, right string) {
 	}
 
 	if errOld == nil && errNew == nil {
-		if old.Version == new.Version {
-			left = string(old.Version) + "-" + red(string(old.Pkgrel))
-			right = string(new.Version) + "-" + green(string(new.Pkgrel))
-		} else {
-			left = red(string(old.Version)) + "-" + string(old.Pkgrel)
-			right = bold(green(string(new.Version))) + "-" + string(new.Pkgrel)
+		oldVersion := string(old.Version) + "-" + string(old.Pkgrel)
+		newVersion := string(new.Version) + "-" + string(new.Pkgrel)
+
+		diffPosition := 0
+
+		checkWord := func(str string, index int, word string) bool {
+			wordLength := len(word)
+			nextIndex := index + 1
+			return (
+				(index < len(str) - wordLength) &&
+				(str[nextIndex:(nextIndex + wordLength)] == word))
+		}
+
+		for index, char := range oldVersion {
+			if (index >= len(newVersion)) || (char != rune(newVersion[index]))  {
+				break
+			}
+
+			if (
+				!(unicode.IsLetter(char) || unicode.IsNumber(char)) ||
+					(
+						(
+							(index == len(oldVersion) - 1) ||
+							(index == len(newVersion) - 1)) &&
+						(len(oldVersion) != len(newVersion))) ||
+					checkWord(oldVersion, index, "rc") ||
```

Comment:
Why not just make checkWord variadic? Would be much cleaner to do `checkWord(oldversion, index, "rc", "pre", "alpha", "beta")`

Reply:
Done, thank you!